camera_control/camera_control/nodes/camera_control_node.py

```diff
@@ -0,0 +1,264 @@
+
+        laser_point_list = []
+        runner_point_list = []
+
+        if self.laser_pub_control:
+            laser_point_list = cv_utils.detect_laser(frames)
+            laser_msg = self.create_pos_data_msg(laser_point_list, frames)
+            self.laser_pos_publisher.publish(laser_msg)
+        else:
+            # If not publishing laser information save the background image
+            self.background_image = curr_image
+        if self.runner_pub_control:
+            runner_point_list = cv_utils.detect_runners(frames, self.background_image)
+            runner_msg = self.create_pos_data_msg(runner_point_list, frames)
+            self.runner_pos_publisher.publish(runner_msg)
+
+        if self.rec_debug_frame:
+            debug_frame = np.copy(curr_image)
+            debug_frame = cv_utils.draw_laser(debug_frame, laser_point_list)
+            debug_frame = cv_utils.draw_runners(debug_frame, runner_point_list)
+            if self.laser_on and self.runner_point is not None:
+                debug_frame = cv2.drawMarker(
+                    debug_frame,
+                    self.runner_point,
+                    (0, 0, 255),
+                    cv2.MARKER_CROSS,
+                    thickness=5,
+                    markerSize=20,
+                )
+            self.rec_debug.write(debug_frame)
+
+    def create_pos_data_msg(self, point_list, frames, timestamp=None):
+        if timestamp is None:
+            timestamp = frames["color"].get_timestamp() / 1000
+        msg = PosData()
+        msg.pos_list = []
+        msg.point_list = []
+        msg.invalid_point_list = []
+        msg.timestamp = timestamp
+        for point in point_list:
+            point_msg = Point()
+            point_msg.x = point[0]
+            point_msg.y = point[1]
+            pos = self.camera.get_pos_location(point[0], point[1], frames)
+            if pos is not None:
+                pos_msg = Pos()
+                pos_msg.x = pos[0]
+                pos_msg.y = pos[1]
+                pos_msg.z = pos[2]
+                msg.pos_list.append(pos_msg)
+                msg.point_list.append(point_msg)
+            else:
+                msg.invalid_point_list.append(point_msg)
+        return msg
+
+    ###Service Calls
+    def get_frame(self, request, response):
+        response.timestamp = self.curr_frames["color"].get_timestamp()
+        color_array = np.asanyarray(self.curr_frames["color"].get_data())
+        depth_array = np.asanyarray(self.curr_frames["depth"].get_data())
+        color_data = list(color_array.flatten())
+        depth_data = list(depth_array.flatten())
+        response.color_frame = UInt8NumpyArray(data=color_data, shape=color_array.shape)
+        response.depth_frame = Int16NumpyArray(data=depth_data, shape=depth_array.shape)
+        return response
+
+    def single_runner_detection(self, request, response):
+        runner_point_list = cv_utils.detect_runner(self.curr_frames)
```

Comment:
detect_runner doesn't exist in cv_utils

Reply:
thatnks for catching that, standardized to `detect_runners` and `detect_lasers`